### PR TITLE
fix(wallet-mobile): navigation weirdness in Send

### PR DESCRIPTION
**Pull request by jorbuedo**
##### Ticket
https://emurgo.atlassian.net/browse/YV-30

### CODE REVIEW

File: apps/wallet-mobile/src/kernel/navigation.tsx [hunk 1]
```diff
@@ -650,66 +649,3 @@ const getFocusedRouteName = (state: Partial<NavigationState> | NavigationState['
...
-        routeNames: newRoutes.map((r) => r.name),
-      }
-      navigation.reset(newState)
-    })
-
-    return () => task.cancel()
-  }, [navigation, initialRouteId, routesToKeep])
-}
-
-export function useOverridePreviousRoute<RouteName extends string>(previousRouteName: RouteName) {
```
Comment (jorbuedo) [inline apps/wallet-mobile/src/kernel/navigation.tsx:686]: It was only used precisely for those Send navigation overrides, and it's buggy which caused the issue.
Turns out adjusting the history state in a hook that executes every render, based on a route name that can change between renders just isn't safe.

File: apps/wallet-mobile/src/features/Send/useCases/ListAmountsToSend/ListAmountsToSendScreen.tsx [hunk 2]
```diff
@@ -76,7 +74,7 @@ export const ListAmountsToSendScreen = () => {
     // use case: redirect to add token screen if there is no token left
     if (selectedTokensCounter === 1) {
       clearSearch()
-      navigateTo.addToken()
+      navigateTo.addToken(true)
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Send/useCases/ListAmountsToSend/ListAmountsToSendScreen.tsx]: @jorbuedo just for readability can you pass it as object? `{sholdPop: boolean}` something like it

File: apps/wallet-mobile/src/kernel/navigation.tsx (hunk 1 above)
Comment (stackchain) [reply apps/wallet-mobile/src/kernel/navigation.tsx:686]: nice catch!

**Review by stackchain — APPROVED** [review]

**Review by banklesss — APPROVED** [review]

File: apps/wallet-mobile/src/features/Send/common/navigation.ts [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
+import {AppRouteNavigation, TxHistoryRouteNavigation} from '../../../kernel/navigation'
 
 export const useNavigateTo = () => {
   const navigation = useNavigation<TxHistoryRouteNavigation & AppRouteNavigation>()
 
   return useRef({
     selectedTokens: () => navigation.navigate('send-list-amounts-to-send'),
-    addToken: () => navigation.navigate('send-select-token-from-list'),
+    addToken: (removePrevious?: boolean) => {
+      if (removePrevious) navigation.pop()
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Send/common/navigation.ts]:
```suggestion
    addToken: ({shouldPopPrevious}: {shouldPopPrevious?: boolean} = {shouldPopPrevious = false}) => {
      if (shouldPopPrevious) navigation.pop()
```

File: apps/wallet-mobile/src/features/Send/useCases/ListAmountsToSend/ListAmountsToSendScreen.tsx (hunk 2 above)
Comment (stackchain) [inline apps/wallet-mobile/src/features/Send/useCases/ListAmountsToSend/ListAmountsToSendScreen.tsx]:
```suggestion
      navigateTo.addToken({shouldPopPrevious: true})
```

**Pull request merged**